enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/webcomponents.ts

```diff
@@ -0,0 +1,41 @@
+// Convert kebab-case attribute names to their camelCase setting keys.
+export const attributeToSettingKey = (attr: string): string =>
+  attr.replace(/-([a-z])/g, (_, c) => c.toUpperCase());
+
+// Naïve parsing of attribute values -> appropriate JS types (number, boolean, string)
+export const parseAttributeValue = (value: string | null): unknown => {
+  if (value === "[object Object]") {
+    console.error(
+      "You are passing an object to `setAttribute`, this is not supported as the browser will call `.toString()` to it, making it `[object Object]`.\nPlease pass a string or set the property directly, eg embed.theme = {...}",
+    );
+  }
+  if (value === null) {
+    return undefined;
+  }
+
+  // Boolean attributes:
+  if (value === "" || value === "true") {
```

Comment:
Coercing `""` string to `true` seems a bit rough to me... Does this help with defining attributes without values like `<metabase-question drills />`, and it gets implicitly turned into `""`?